test/unit/cmap/connection.test.ts

```diff
@@ -362,81 +389,324 @@ describe('new Connection()', function () {
           connection.onMessage(message);
         });
 
-        it('calls all operation description callbacks with an error', function () {
+        it('calls all operation description callbacks with an error', function (done) {
           expect(spyOne).to.be.calledOnce;
           expect(spyTwo).to.be.calledOnce;
           const errorOne = spyOne.firstCall.args[0];
           const errorTwo = spyTwo.firstCall.args[0];
           expect(errorOne).to.be.instanceof(MongoRuntimeError);
           expect(errorTwo).to.be.instanceof(MongoRuntimeError);
+          done();
         });
       });
     });
   });
 
-  describe('onTimeout()', () => {
+  describe('when the socket times out', () => {
     let connection: sinon.SinonSpiedInstance<Connection>;
     let clock: sinon.SinonFakeTimers;
     let timerSandbox: sinon.SinonFakeTimers;
-    let driverSocket: sinon.SinonSpiedInstance<FakeSocket>;
+    let driverSocket: FakeSocket;
     let messageStream: MessageStream;
-    let kDelayedTimeoutId: symbol;
+    let callbackSpy;
+    let closeCount = 0;
+    let kDelayedTimeoutId;
     let NodeJSTimeoutClass: any;
 
     beforeEach(() => {
       timerSandbox = createTimerSandbox();
-      clock = sinon.useFakeTimers();
+      clock = sinon.useFakeTimers({
+        toFake: ['nextTick', 'setTimeout', 'clearTimeout']
+      });
+      driverSocket = new FakeSocket();
+      // @ts-expect-error: driverSocket does not fully satisfy the stream type, but that's okay
+      connection = new Connection(driverSocket, connectionOptionsDefaults);
+      const messageStreamSymbol = getSymbolFrom(connection, 'messageStream');
+      messageStream = connection[messageStreamSymbol];
 
+      callbackSpy = sinon.spy();
+      // Create the operation description.
+      const operationDescription: OperationDescription = {
+        requestId: 1,
+        cb: callbackSpy
+      };
+
+      connection.on('close', () => {
+        closeCount++;
+      });
+
+      // Stick an operation description in the queue.
+      const queueSymbol = getSymbolFrom(connection, 'queue');
+      connection[queueSymbol].set(1, operationDescription);
+
+      kDelayedTimeoutId = getSymbolFrom(connection, 'delayedTimeoutId');
       NodeJSTimeoutClass = setTimeout(() => null, 1).constructor;
+    });
 
-      driverSocket = sinon.spy(new FakeSocket());
+    afterEach(() => {
+      closeCount = 0;
+      sinon.restore();
+      timerSandbox.restore();
+      clock.restore();
+    });
+
+    context('delayed timeout for lambda behavior', () => {
+      let cleanupSpy;
+      let timeoutSpy;
+      beforeEach(() => {
+        cleanupSpy = sinon.spy(connection, '_cleanup');
+        timeoutSpy = sinon.spy(connection, 'onTimeout');
+      });
+
+      afterEach(() => {
+        sinon.restore();
+        timerSandbox.restore();
+        clock.restore();
+        sinon.restore();
+      });
```

Comment:
dupe clean up, double sinon.restore and overlap with the above afterEach, I assume this is left from debuggin and not intentional?